### PR TITLE
Add ocean resolution to setup_expt invocation and retire/reduce COMROT/ROTDIR usage

### LINKED ISSUE #2061: Clean out COMROT and reduce ROTDIR usage
### What new functionality do you need?

Remove the `COMROT` variable (replace with `ROTDIR`) and reduce `ROTDIR` variable where it still exists (as able).

### What are the requirements for the new functionality?

Reduces usage of `COMROT` and `ROTDIR` variable further.

### Acceptance Criteria

System does not break and existing functionality remains intact.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR combines two issues, both of which make updates to the global-workflow setup API.

Firstly, this PR does some housekeeping for issue #2061:

1. Removes the `COMROT` variable, replaces it with `ROTDIR` or other appropriate variable(s), and reduces the usage of `ROTDIR` where able. Some instances of `ROTDIR` remain since they are necessary still. Some updates were made to scripts that may be removed or rewritten in the future; changes were still made for consistency and to remove `COMROT`/`ROTDIR`.
2. The hosts files are updated to remove `COMROOT` and move the values into the `COMINsyn` settings (which is where they were used anyway). Need to free up `COMROOT` to be filled by the `--comroot` setup argument.
3. Additionally, several instances of `COMROT` were changed to be `COMROOT` or `comroot` (2nd "O"/"o" added). The use of `--comroot` as an argument for the setup scripts and `COMROOT` to define `ROTDIR` in `config.base` brings the system more in line with how NCO uses `COMROOT` in operations to define the location of com (similar to where developers place their `ROTDIR`s. This is a subtle and easy to miss update for the setup script arguments (`comrot` -> `comroot`). An announcement is recommended so users begin using `--comroot` instead of `--comrot` with the setup scripts.
4. Documentation is also updated

Secondly, this PR allows a user to provide a desired ocean resolution during the invocation of setup_expt.py:

1. renames `--resdet` option to `--resdetatmos`
2. adds `--resdetocean`. If a value `<= 0`. is provided, the program will choose based on pre-selected options.
3. updates the CI test yamls
5. updates the documentation

Assigning @TerrenceMcGuinness-NOAA as a reviewer since this PR touches the CI files.

Resolves #2061

This PR has been updated to merge PR #2204 into it.

# Type of change

- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? No, but users will need to update their use of setup_expt.py
- Does this change require a documentation update? YES

# How has this been tested?

- Created a few of the experiment directories using `create_experiment.py` and the various `ci/cases` yaml files and inspected `config.base` in the respective experiment directories.
- Cycled atmos-only C96C48 test on Hera.
- Forecast-only S2SW C48 test on Hera.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by KateFriedman-NOAA** [comment]
The forecast-only S2SW C48 test on Hera is still running at the time of initiating this PR. The IC and prep jobs have completed successfully though.

File: jobs/JGFS_ATMOS_FSU_GENESIS
Comment (WalterKolczynski-NOAA) [inline jobs/JGFS_ATMOS_FSU_GENESIS:44]: - Does the genesis script use all of `COMIN`, `gfsdir`, and `COMINgfs`?
- `gdasdir` (and `COMINgdas`) are no longer being defined.

File: jobs/JGLOBAL_ATMOS_POST_MANAGER
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ATMOS_POST_MANAGER]: I don't think this second one is needed
```suggestion
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: jobs/JGFS_ATMOS_FSU_GENESIS
Comment (KateFriedman-NOAA) [reply jobs/JGFS_ATMOS_FSU_GENESIS:44]: > * Does the genesis script use all of `COMIN`, `gfsdir`, and `COMINgfs`?

Yes, there is a mixture of all three across both the `TC_tracker` (dev) and `ens_tracker` (ops) packages. Note, the FSU_GENESIS job is not tested so further changes to the JJOB script are potentially needed.

> * `gdasdir` (and `COMINgdas`) are no longer being defined.

The FSU genesis job is only run for `gfs` in operations so we don't need the `gdas` versions of those path variables.

**Comment by KateFriedman-NOAA** [comment]
> create_experiment.py: error: unrecognized arguments: --rotdir /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/RUNTESTS/ROTDIR
> ROOT/PR/2214/RUNTESTS/ROTDIR --expdir /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/RUNTESTS/EXPDIR --idate 2021032312 --edate 2021032312 --yaml /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/global-workflow/ci/platforms/gfs_defaults_ci.yaml
>

I see the problem and how to fix. Will fix and retry CI tests.

**Review by TerrenceMcGuinness-NOAA — APPROVED** [review]
Nice work, so cool.  So glad to get rid of the ROT stuff!  Looking forward to studying the updated documentation to finally get a handle on the staging directories.

**Comment by KateFriedman-NOAA** [comment]
Killed CI tests again after receiving request to merge branch from PR #2204 into branch for this PR. Will merge branches/PRs and then restart CI tests.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI testing.

**Review by TerrenceMcGuinness-NOAA — APPROVED** [review]
changes to CI scripts look great.  All substantive updates in the CI Bash scrips pertain to creating experiment which is ran via the PR during CI run-time so no updates to the In-Place cron scripts are needed to test.

**Pull request merged**
